photon-api/src/main/scala/com/linkedin/photon/ml/function/ObjectiveFunctionHelper.scala

```diff
@@ -36,11 +36,11 @@ object ObjectiveFunctionHelper {
    * @return A function which builds the appropriate type of [[ObjectiveFunction]] for a given [[Coordinate]] type and
    *         optimization settings.
    */
-  def buildFactory(taskType: TaskType, treeAggregateDepth: Int): ObjectiveFunctionFactory =
+  def buildFactory(taskType: TaskType, treeAggregateDepth: Int)(interceptIndex: Int = -1): ObjectiveFunctionFactory =
```

Comment:
Don't need to curry here - these parameters are all available at the same time:
```suggestion
  def buildFactory(taskType: TaskType, treeAggregateDepth: Int, interceptIndexOpt: Option[Int] = None): ObjectiveFunctionFactory =
```

Reply:
I think we still need to keep curry here. `lossFunctionFactory` was created outside of `coordinates` (L658 in `GameEstimator.scala`) and `lossFunctionFactory` takes different `interceptIndex` each time while generating `coordinates` in `GameEstimator.scala`. Do you have any other suggestions?